services/fileuploader/fileuploader.go

```diff
@@ -121,13 +155,30 @@ func (p *provider) updateLoop(ctx context.Context, backendConfig backendconfig.B
 			if bucket.Type != "" && len(bucket.Config) > 0 {
 				preferences = c.Settings.DataRetention.StoragePreferences
 			}
-			settings[workspaceId] = StorageSettings{
+
+			// if no change in storage configuration, don't create new Filemanager
+			if ok && reflect.DeepEqual(currentWorkspaceConfig.Bucket, bucket) && reflect.DeepEqual(currentWorkspaceConfig.Preferences, preferences) {
+				settingsMap[workspaceId] = currentWorkspaceConfig
+				filemanagerMap[workspaceId] = currentFileManagerMap[workspaceId]
+				continue
+			}
+
+			// either newly polled workspace settings or updated storage config - update object storage Filemanager
+			settingsMap[workspaceId] = StorageSettings{
 				Bucket:      bucket,
 				Preferences: preferences,
+				updatedAt:   time.Now(),
 			}
+			filemanagerMap[workspaceId] = sync.OnceValues(func() (filemanager.FileManager, error) {
+				return filemanager.New(&filemanager.Settings{
+					Provider: bucket.Type,
+					Config:   bucket.Config,
+				})
+			})
 		}
 		p.mu.Lock()
-		p.storageSettings = settings
+		p.storageSettings = settingsMap
+		p.fileManagerMap = filemanagerMap
```

Comment:
When overwriting here, it could be that some file managers will be free for garbage collection, right? Could it be that these file managers have dangling connections that might need cleaning up first? I see no `Close()` method in the interface so I'm talking about what's going on under the hood.